packages/graphql-transformer-core/src/util/sanity-check.ts

```diff
@@ -271,6 +271,46 @@ export function cantHaveMoreThan200Resources(diffs: Diff[], currentBuild: Diffab
   }
 }
 
+/**
+ * Throws a helpful error when a customer is trying to complete an invalid migration.
+ * Users are unable to add multiple GSIs at the same time in update stage.
+ * @param diffs The set of diffs between currentBuild and nextBuild.
+ * @param currentBuild The last deployed build.
+ * @param nextBuild The next build.
+ */
+export function cantAddMultipleGSIAtUpdateTime(diffs: Diff[], currentBuild: DiffableProject, nextBuild: DiffableProject) {
```

Comment:
This can also happen when deleting more than one GSI. This could be changed to something like `cantMutateMutltipleGSIs`